File: Python/Course.py
from Dosen import Dosen
from Mahasiswa import Mahasiswa

class Course:
    __matkul = ""
    __dosen = ""
    

    def __init__(self, matkul = "", dosen = Dosen(), mahasiswa = Mahasiswa()):
        self.__mahasiswa = []
        self.__matkul = matkul
        self.__dosen = dosen
        self.__mahasiswa.append(mahasiswa)
    
    # setter getter matkul
    def setMatkul(self, matkul):
        self.__matkul = matkul

    def getMatkul(self):
        return self.__matkul
    
    # setter getter dosen
    def setDosen(self, dosen):
        self.__dosen = dosen

    def getDosen(self):
        return self.__dosen
    
    # setter getter mahasiswa
    def tambah_Mahasiswa(self, mahasiswa):
        self.__mahasiswa.append(mahasiswa)
    
    def getMahasiswa(self):
        return self.__mahasiswa
    
    
    # Method
    def printListCourse(self):
        print("Mata Kuliah : " + self.__matkul)
        self.__dosen.printDosen()
        print("")
        print("")
        print("Daftar Mahasiswa: ")
        for coba in self.__mahasiswa:
            coba.printMahasiswa()
            print("")
            
    
    def printCourse(self):
        print("Mata Kuliah : " + self.__matkul)

File: C++/Course.cpp
#include "header.hh"

/* = => COURSE SECTION <= = */

/* KONSTRUKTOR */
Course::Course(){
    matkul = "";
}

Course::Course(string matkul, Dosen dosen, Mahasiswa murid){
    this->matkul = matkul;
    this->dosen = dosen;
    this->llistMurid.push_back(murid);
}

void Course::setMatkul(string matkul){
    this->matkul = matkul;
}

string Course::getMatkul(){
    return this->matkul;
}

void Course::setDosen(Dosen dosen){
    this->dosen = dosen;
}

Dosen Course::getDosen(){
    return this->dosen;
}

void Course::addMahasiswa(Mahasiswa murid){
    this->llistMurid.push_back(murid);
}

void Course::printListCourse(){
    cout << "Mata Kuliah            :" << this->matkul << endl;
    cout << "       Deskripsi Dosen     " << endl;
    cout << "===========================" << endl;
    this->dosen.printDosen();
    cout << "       Profil Mahasiswa     " << endl;
    cout << "============================" << endl;
    for(list<Mahasiswa>::iterator it = llistMurid.begin(); it != llistMurid.end(); it++){
        it->printMahasiswa();
        cout << '\n';
    }
}

void Course::printCourse(){
    cout << "Mata Kuliah    : " << this->matkul << endl;
}

Course::~Course(){

}

